Sources/PostgreSQL/Connection/PostgreSQLConnection+TCP.swift

```diff
@@ -19,8 +21,12 @@ extension PostgreSQLConnection {
                 }
         }
 
-        return bootstrap.connect(host: hostname, port: port).map(to: PostgreSQLConnection.self) { channel in
-            return .init(queue: handler, channel: channel)
+        return bootstrap.connect(host: hostname, port: port).flatMap(to: PostgreSQLConnection.self) { channel in
+            let connection = PostgreSQLConnection(queue: handler, channel: channel)
+            if let tlsConfiguration = tlsConfiguration {
+                return connection.addSSLClientHandler(using: tlsConfiguration).transform(to: connection)
+            }
+            return Future.map(on: worker) { connection }
```

Comment:
We should move to using the more performant `worker.eventLoop.newSucceededFuture()` method.